frontend/src/v5/ui/components/shared/modalsDispatcher/templates/alertModal/alertModal.component.tsx

```diff
@@ -37,30 +37,32 @@ interface IAlertModal {
 export const AlertModal: FC<IAlertModal> = ({ onClickClose, currentActions = '', error, details, open }) => {
 	const teamspace = TeamspacesHooksSelectors.selectCurrentTeamspace();
 	const project = ProjectsHooksSelectors.selectCurrentProject();
+	const accessibleProjects = ProjectsHooksSelectors.selectProjects()[teamspace] || [];
+	const hasAccessToProject = accessibleProjects.findIndex(({ _id }) => _id === project) >= 0;
 	const history = useHistory();
 
 	const message = getErrorMessage(error);
 	const code = getErrorCode(error);
 	const status = getErrorStatus(error);
 	const errorStatus = status && code ? `${status} - ${code}` : '';
 	const pathNotFound = isPathNotFound(error);
+	const modelNotFound = isModelNotFound(code);
+	const projectNotFound = isProjectNotFound(code);
 	const unauthorized = isPathNotAuthorized(error);
-	const unauthInTeamspace = unauthorized && teamspace;
-	const unauthInProject = unauthorized && project;
 
 	const getSafePath = () => {
 		// eslint-disable-next-line max-len
-		if (isResourceNotFound(code) || (unauthInProject)) return generatePath(PROJECT_ROUTE_BASE, { teamspace, project });
-		if (isProjectNotFound(code) || (unauthInTeamspace)) return generatePath(TEAMSPACE_ROUTE_BASE, { teamspace });
+		if ((modelNotFound || unauthorized) && hasAccessToProject) return generatePath(PROJECT_ROUTE_BASE, { teamspace, project });
```

Comment:
this logic seems a bit complex here.  What does it mean?

Reply:
It basically checks whether something (a model/a project, etc) is not reachable and where the user can be redirected to.

For instance, in the line above, if the user does not have access to a model (lack of authorisation or model not found), but they have access to the project, this is where they get redirected